org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/AbstractAnnotatedMethodFilter.java

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2017 Mountainminds GmbH & Co. KG and Contributors
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Marc R. Hoffmann - initial API and implementation
+ *
+ *******************************************************************************/
+package org.jacoco.core.internal.analysis.filter;
+
+import java.util.List;
+
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.MethodNode;
+
+/**
+ * Filters which filters annoted methods.
+ */
+public abstract class AbstractAnnotatedMethodFilter implements IFilter {
+	private final String descType;
+
+	/**
+	 * @param annotationType
+	 */
+	protected AbstractAnnotatedMethodFilter(final String annotationType) {
+		this.descType = "L" + annotationType + ";";
+	}
+
+	public void filter(final String className, final String superClassName,
+			final MethodNode methodNode, final IFilterOutput output) {
+		if (hasAnnotation(methodNode)) {
+			output.ignore(methodNode.instructions.getFirst(),
+					methodNode.instructions.getLast());
+		}
+	}
+
+	private boolean hasAnnotation(final MethodNode methodNode) {
+		final List<AnnotationNode> runtimeInvisibleAnnotations = methodNode.invisibleAnnotations;
```

Comment:
In contrast to Lombok, Groovy adds visible annotation, while here we check only `invisibleAnnotations`, so that currently filter doesn't work.

Reply:
@Godin Excellent catch! Fixed ✅

Reply:
@marchof nothing extraordinary - I'm simply doing end-to-end testing ;)